tabs/pid_tuning.js

```diff
@@ -687,35 +692,34 @@ TABS.pid_tuning.initialize = function (callback) {
         $('a.update').click(function () {
             form_to_pid_and_rc();
 
-            var promise;
-            if (semver.gte(CONFIG.apiVersion, CONFIGURATOR.pidControllerChangeMinApiVersion)) {
-                PID.controller = pidController_e.val();
-                promise = MSP.promise(MSP_codes.MSP_SET_PID_CONTROLLER, MSP.crunch(MSP_codes.MSP_SET_PID_CONTROLLER));
-            } else {
-                promise = new Promise(function (resolve) {
-                    resolve();
-                });
-            }
-            promise.then(function () {
+            Promise.resolve(true)
+            .then(function () {
+                var promise;
+                if (semver.gte(CONFIG.apiVersion, CONFIGURATOR.pidControllerChangeMinApiVersion)) {
+                    PID.controller = pidController_e.val();
+                    promise = MSP.promise(MSP_codes.MSP_SET_PID_CONTROLLER, MSP.crunch(MSP_codes.MSP_SET_PID_CONTROLLER));
+                }
+                return promise;
+            }).then(function () {
                 return MSP.promise(MSP_codes.MSP_SET_PID, MSP.crunch(MSP_codes.MSP_SET_PID));
             }).then(function () {
-                return MSP.promise(MSP_codes.MSP_SET_PID_ADVANCED, MSP.crunch(MSP_codes.MSP_SET_PID_ADVANCED));
+                if (semver.gte(CONFIG.flightControllerVersion, "2.9.0") && semver.lt(CONFIG.flightControllerVersion, "3.0.0")) {
+                  return MSP.promise(MSP_codes.MSP_SET_SPECIAL_PARAMETERS, MSP.crunch(MSP_codes.MSP_SET_SPECIAL_PARAMETERS));
+                }
+            }).then(function () {
+                if (semver.gte(CONFIG.flightControllerVersion, "2.8.2")) {
+                  return MSP.promise(MSP_codes.MSP_SET_PID_ADVANCED, MSP.crunch(MSP_codes.MSP_SET_PID_ADVANCED));
+                }
             }).then(function () {
-                var promise;
                 if (semver.gte(CONFIG.flightControllerVersion, "2.8.1")) {
-                    promise = MSP.promise(MSP_codes.MSP_SET_FILTER_CONFIG, MSP.crunch(MSP_codes.MSP_SET_FILTER_CONFIG));
+                    return MSP.promise(MSP_codes.MSP_SET_FILTER_CONFIG, MSP.crunch(MSP_codes.MSP_SET_FILTER_CONFIG));
                 }
-
-                return promise;
```

Comment:
The 'easier to read' is a matter of preference. I personally find it easier, when looking at one location in a long method, if I don't have to worry about the code path I am interested in never reaching there because one of potentially a number of early exits has been hit. This is probably biased by me preferring maintainability over 'easy on the eye', and thus always making the assumption that no matter how short a method is, it will sooner or later grow to become unwieldy.